frontend/src/features/undo_redo/store/MapHistoryStore.ts

```diff
@@ -122,6 +140,36 @@ function handleAddObjectAction(state: MapState, action: ObjectAddAction): MapSta
   };
 }
 
+function handlePasteClipboardAction(state: MapState, clipboard: MapState): MapState {
+  const newLayers = { ...state.layers };
+
+  clipboard.layers.plant.objects.forEach((object) => {
+    const layerIndex = object.index;
+    const newObject = { ...object };
```

Comment:
The id would have to change somehow. Imo handling it here seems better than in the copy action in the component.